src/lib.rs

```diff
@@ -238,15 +238,22 @@ impl<B: MysqlShim<W>, R: Read, W: Write> MysqlIntermediary<B, R, W> {
     fn init(&mut self) -> Result<(), B::Error> {
         self.writer.write_all(&[10])?; // protocol 10
 
+        let mut capabilities = CapabilityFlags::empty();
+        capabilities.insert(CapabilityFlags::CLIENT_PROTOCOL_41);
+        capabilities.insert(CapabilityFlags::CLIENT_RESERVED);
```

Comment:
What does setting this do? Aren't generally reserved flags supposed to be 0?

Reply:
It came from default capabilities flags. Here you set it.
https://github.com/jonhoo/msql-srv/blob/4f301c9757954fcef3961ce4a50c8d469edd8b6f/src/lib.rs#L246
It should be `0x02` if you only meant `CLIENT_PROTOCOL_41`.